packages/verification-core/src/verifier/CertifactionEthVerifier.js

```diff
@@ -48,49 +43,42 @@ export default class CertifactionEthVerifier {
             eth,
             legacyContract,
             fallbackLegacyContracts,
-            claimContract,
+            claimContract
+        )
+        this.certifactionClaimVerifier = new CertifactionClaimVerifier(
             acceptedIssuerKey,
+            this.certifactionEthClient,
             certifactionApiUrl
         )
     }
 
-    readPdfBytes(file) {
-        return new Promise((resolve, reject) => {
-            const reader = new FileReader()
-
-            reader.onload = () => {
-                resolve(new Uint8Array(reader.result))
-            }
-
-            reader.onerror = (error) => {
-                reject(error)
-            }
-
-            reader.readAsArrayBuffer(file)
-        })
-    }
-
-    async verify(file) {
-        const pdfBytes = await this.readPdfBytes(file)
-        const [fileHash, encryptionKeys] = await Promise.all([
-            hashingService.hashFile(pdfBytes),
-            this.pdfService.extractEncryptionKeys(pdfBytes)
-        ])
-
+    /**
+     * Verify the given file hash using claims and legacy contract
+     *
+     * @param {string} fileHash
+     * @param {string} decryptionKey
+     *
+     * @returns {Promise<Object>}
+     */
+    async verify(fileHash, decryptionKey) {
         let verification = {
             hash: fileHash
         }
 
         try {
-            let fileVerification
+            let fileVerification = null
 
             if (this.enableClaims) {
                 console.log('Verifying with claim method...')
-                fileVerification = await this.certifactionEthClient.verifyFile(
-                    fileHash,
-                    (encryptionKeys !== null) ? encryptionKeys.privateKey : null
-                )
-            } else {
+                fileVerification = await this.certifactionClaimVerifier.verify(fileHash, decryptionKey)
+            }
+
+            if (fileVerification === null) {
+                if (this.enableClaims) {
+                    // If the claim verifier doesn't return a result, try verifying by legacy contracts
+                    console.log('No claims found, fallback to legacy contract based verification')
```

Comment:
It would also be possible to have `if (fileVerification === null)` directly after `certifactionClaimVerifier.verify()`, but either checking if the claims are enabled or checking if fileVerification is null again is necessary. I think it's discussable if we even still need the feature flag enableClaims. @emeka what do you think, should I remove it?

Reply:
Claims is the default now.

Reply:
Yes, I think we don't need the feature flag anymore, I will remove it.